### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by mrBliss**
Note that this does not incorporate the latest changes in cardano-ledger-specs.

Notable changes:
* https://github.com/input-output-hk/ouroboros-network/pull/2807
* https://github.com/input-output-hk/ouroboros-network/pull/2811
* https://github.com/input-output-hk/ouroboros-network/pull/2832

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 1]
```diff
@@ -857,6 +858,7 @@ nodeToNodeTracers' trSel verb tr =
   , NodeToNode.tBlockFetchTracer = tracerOnOff (traceBlockFetchProtocol trSel) verb "BlockFetchProtocol" tr
   , NodeToNode.tBlockFetchSerialisedTracer = showOnOff (traceBlockFetchProtocolSerialised trSel) "BlockFetchProtocolSerialised" tr
   , NodeToNode.tTxSubmissionTracer = tracerOnOff (traceTxSubmissionProtocol trSel) verb "TxSubmissionProtocol" tr
+  , NodeToNode.tTxSubmission2Tracer = nullTracer -- TODO @coot: tracerOnOff (traceTxSubmission2Protocol trSel) verb "TxSubmission2Protocol" tr
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:861]: Draft because this still needs to be done by somebody from the network team. cc @coot.

**Review by deepfire — APPROVED** [review]
LGTM, :heart:

File: cardano-node/src/Cardano/Tracing/Tracers.hs (hunk 1 above)
Comment (mrBliss) [reply cardano-node/src/Cardano/Tracing/Tracers.hs]: Thanks @coot! Shall I squash before merging?

Comment (coot) [reply cardano-node/src/Cardano/Tracing/Tracers.hs]: Done, I pushed two commits to this branch.

**Comment by coot** [comment]
I found a typo in my commit, I am going to `git push --force`.

**Comment by mrBliss** [comment]
> I found a typo in my commit, I am going to `git push --force`.

Go ahead. You can even squash everything in my first commit.

**Comment by coot** [comment]
> Go ahead. You can even squash everything in my first commit.

Done

**Comment by coot** [comment]
The windows build fails:
```
cabal.exe: Could not resolve dependencies:
[__0] trying: byron-spec-chain-0.1.0.0 (user goal)
[__1] trying: hedgehog-1.0.4 (dependency of byron-spec-chain)
[__2] trying: cardano-api-1.24.2 (user goal)
[__3] rejecting: cardano-api:!test (constraint from config file, command line
flag, or user target requires opposite flag selection)
[__3] trying: cardano-api:*test
[__4] next goal: tasty-hedgehog (dependency of cardano-api *test)
[__4] rejecting: tasty-hedgehog-1.0.0.2 (conflict: hedgehog==1.0.4,
tasty-hedgehog => hedgehog>=1.0.2 && <1.0.4)
[__4] skipping: tasty-hedgehog-1.0.0.1, tasty-hedgehog-1.0.0.0,
tasty-hedgehog-0.2.0.0, tasty-hedgehog-0.1.0.2, tasty-hedgehog-0.1.0.1,
tasty-hedgehog-0.1.0.0 (has the same characteristics that caused the previous
version to fail: excludes 'hedgehog' version 1.0.4)
[__4] fail (backjumping, conflict set: cardano-api, hedgehog, tasty-hedgehog,
cardano-api:test)
Backjump limit reached (currently 4000, change with --max-backjumps or try to
run with --reorder-goals).
```
`tasty-hedgehog-1.0.0.2` is needed to complete this plan, it was published on `16.12.2020`, updating `index-state` might help as its  a few days older:
```
'hackage.haskell.org'! Falling back to older state (2020-12-10T21:37:13Z).
```

**Review by erikd — APPROVED** [review]

**Comment by hamishmack** [comment]
Added the `hydra-dont-build` label as the eval keeps running out of memory.  We think #2285 will fix the issue with too many jobs being included.

**Comment by kderme** [comment]
Could we have this pr in the update?

**Comment by kderme** [comment]
bors merge

**Pull request merged**